diff --git a/src/harfbuzz-icu.pc.in b/src/harfbuzz-icu.pc.in
index 949869a..028a564 100644
--- a/src/harfbuzz-icu.pc.in
+++ b/src/harfbuzz-icu.pc.in
@@ -8,6 +8,7 @@ Description: HarfBuzz text shaping library ICU integration
 Version: %VERSION%
 
 Requires: harfbuzz
-Requires.private: icu-uc
-Libs: -L${libdir} -lharfbuzz-icu
+Requires.private: icu-uc cairo-ft glib-2.0
+Libs: -L${libdir} -lfreetype -lharfbuzz-icu
+Libs.private: -lusp10 -lgdi32
 Cflags: -I${includedir}/harfbuzz
diff --git a/src/harfbuzz.pc.in b/src/harfbuzz.pc.in
index 661251c..a423b0a 100644
--- a/src/harfbuzz.pc.in
+++ b/src/harfbuzz.pc.in
@@ -7,7 +7,8 @@ Name: harfbuzz
 Description: HarfBuzz text shaping library
 Version: %VERSION%
 
-Libs: -L${libdir} -lharfbuzz
-Libs.private: -lm %libs_private%
+Requires.private: cairo-ft glib-2.0
+Libs: -L${libdir} -lfreetype -lharfbuzz
+Libs.private: -lm %libs_private% -lusp10 -lgdi32
 Requires.private: %requires_private%
 Cflags: -I${includedir}/harfbuzz
