diff --git a/CMakeLists.txt b/CMakeLists.txt
index 92acc73..65ac95b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -418,6 +418,7 @@ ELSEIF(CMAKE_C_COMPILER_ID MATCHES "MSVC")
 ENDIF()
 SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${WARNING_AS_ERROR}")
 ENDIF()
+SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${WARNING_AS_ERROR}")
 
 
 IF(NOT REMOTEIO_PLUGIN_TYPE MATCHES "OFF")
diff --git a/include/mysql.h b/include/mysql.h
index ce633e4..cd0cf91 100644
--- a/include/mysql.h
+++ b/include/mysql.h
@@ -516,7 +516,7 @@ typedef struct
 } MARIADB_X509_INFO;
 
 
-struct st_mysql_client_plugin *
+struct st_mysql_client_plugin * STDCALL
 mysql_load_plugin(struct st_mysql *mysql, const char *name, int type,
                   int argc, ...);
 struct st_mysql_client_plugin * STDCALL
diff --git a/include/mysql/client_plugin.h b/include/mysql/client_plugin.h
index 6c9d178..8ef2d22 100644
--- a/include/mysql/client_plugin.h
+++ b/include/mysql/client_plugin.h
@@ -203,7 +203,7 @@ typedef struct st_mysql_client_plugin_REMOTEIO
   @retval
   a pointer to the loaded plugin, or NULL in case of a failure
 */
-struct st_mysql_client_plugin *
+struct st_mysql_client_plugin * STDCALL
 mysql_load_plugin(struct st_mysql *mysql, const char *name, int type,
                   int argc, ...);
 
diff --git a/libmariadb/CMakeLists.txt b/libmariadb/CMakeLists.txt
index 7589850..153d262 100644
--- a/libmariadb/CMakeLists.txt
+++ b/libmariadb/CMakeLists.txt
@@ -492,20 +492,37 @@ SET_TARGET_PROPERTIES(libmariadb PROPERTIES VERSION
  ${CPACK_PACKAGE_VERSION_MAJOR}
  SOVERSION ${CPACK_PACKAGE_VERSION_MAJOR})
 
+IF(MINGW)
+  SET_TARGET_PROPERTIES(libmariadb PROPERTIES
+                        OUTPUT_NAME mariadb
+                        RUNTIME_OUTPUT_NAME libmariadb
+                        ARCHIVE_OUTPUT_NAME mariadb)
+ENDIF()
+
 IF(NOT WIN32)
   SET_TARGET_PROPERTIES(mariadbclient PROPERTIES OUTPUT_NAME "${LIBMARIADB_STATIC_NAME}")
 ENDIF()
 
 INSTALL(TARGETS mariadbclient
-          COMPONENT Development
-          DESTINATION ${INSTALL_LIBDIR})
+        COMPONENT Development
+        RUNTIME DESTINATION ${INSTALL_BINDIR}
+        LIBRARY DESTINATION ${INSTALL_LIBDIR}
+        ARCHIVE DESTINATION ${INSTALL_LIBDIR}
+        )
 IF(WIN32)
 INSTALL(TARGETS libmariadb
         COMPONENT SharedLibraries
-        DESTINATION ${INSTALL_LIBDIR})
+        RUNTIME DESTINATION ${INSTALL_BINDIR}
+        LIBRARY DESTINATION ${INSTALL_LIBDIR}
+        ARCHIVE DESTINATION ${INSTALL_LIBDIR}
+        )
 ELSE()
-INSTALL(TARGETS libmariadb LIBRARY DESTINATION ${INSTALL_LIBDIR}
-        COMPONENT SharedLibraries NAMELINK_COMPONENT Development)
+INSTALL(TARGETS libmariadb
+        COMPONENT SharedLibraries NAMELINK_COMPONENT Development
+        RUNTIME DESTINATION ${INSTALL_BINDIR}
+        LIBRARY DESTINATION ${INSTALL_LIBDIR}
+        ARCHIVE DESTINATION ${INSTALL_LIBDIR}
+        )
 ENDIF()
 
 IF(MSVC)
diff --git a/libmariadb/secure/schannel_certs.c b/libmariadb/secure/schannel_certs.c
index baec802..18ae6c2 100644
--- a/libmariadb/secure/schannel_certs.c
+++ b/libmariadb/secure/schannel_certs.c
@@ -65,7 +65,7 @@ static DWORD get_last_error()
 #define FAIL(...) \
    do{\
      status = get_last_error();\
-     ma_format_win32_error(errmsg, errmsg_len, status, __VA_ARGS__);\
+     ma_format_win32_error(errmsg, errmsg_len, status, ##__VA_ARGS__);\
      goto cleanup;\
   } while (0)
 
