diff --git a/include/ma_global.h b/include/ma_global.h
index b8ef7e1..77865ca 100644
--- a/include/ma_global.h
+++ b/include/ma_global.h
@@ -33,9 +33,9 @@
 #if _MSC_VER < 1900
 #define snprintf _snprintf
 #endif
-#define strerror_r(errno,buf,len) strerror_s(buf,len,errno)
 #endif
 #define STDCALL __stdcall 
+#define strerror_r(errno,buf,len) strerror_s(buf,len,errno)
 #endif
 
 #include <ma_config.h>
diff --git a/include/mysql.h b/include/mysql.h
index ba3063d..a19f081 100644
--- a/include/mysql.h
+++ b/include/mysql.h
@@ -473,7 +473,7 @@ struct st_mysql_client_plugin
   MYSQL_CLIENT_PLUGIN_HEADER
 };
 
-struct st_mysql_client_plugin *
+struct st_mysql_client_plugin * STDCALL
 mysql_load_plugin(struct st_mysql *mysql, const char *name, int type,
                   int argc, ...);
 struct st_mysql_client_plugin * STDCALL
diff --git a/include/mysql/client_plugin.h b/include/mysql/client_plugin.h
index 262fd4a..9a0e1e6 100644
--- a/include/mysql/client_plugin.h
+++ b/include/mysql/client_plugin.h
@@ -194,7 +194,7 @@ typedef struct st_mysql_client_plugin_REMOTEIO
   @retval
   a pointer to the loaded plugin, or NULL in case of a failure
 */
-struct st_mysql_client_plugin *
+struct st_mysql_client_plugin * STDCALL
 mysql_load_plugin(struct st_mysql *mysql, const char *name, int type,
                   int argc, ...);
 
diff --git a/libmariadb/CMakeLists.txt b/libmariadb/CMakeLists.txt
index 070fdc9..23e742e 100644
--- a/libmariadb/CMakeLists.txt
+++ b/libmariadb/CMakeLists.txt
@@ -487,6 +487,13 @@ SET_TARGET_PROPERTIES(libmariadb PROPERTIES VERSION
  ${CPACK_PACKAGE_VERSION_MAJOR}
  SOVERSION ${CPACK_PACKAGE_VERSION_MAJOR})
 
+IF(MINGW)
+  SET_TARGET_PROPERTIES(libmariadb PROPERTIES
+                        OUTPUT_NAME mariadb
+                        RUNTIME_OUTPUT_NAME libmariadb
+                        ARCHIVE_OUTPUT_NAME mariadb)
+ENDIF()
+
 IF(NOT WIN32)
   SET_TARGET_PROPERTIES(mariadbclient PROPERTIES OUTPUT_NAME "${LIBMARIADB_STATIC_NAME}")
 ENDIF()
@@ -503,9 +510,15 @@ ELSE()
 #INSTALL(TARGETS libmariadb LIBRARY DESTINATION ${INSTALL_LIBDIR}
 #        COMPONENT SharedLibraries NAMELINK_COMPONENT Development)
 # but as long as we build on CentOS 7 with its cmake 2.8.12.2 we have to use
-INSTALL(TARGETS libmariadb LIBRARY DESTINATION ${INSTALL_LIBDIR}
+INSTALL(TARGETS libmariadb
+        RUNTIME DESTINATION ${INSTALL_BINDIR}
+        LIBRARY DESTINATION ${INSTALL_LIBDIR}
+        ARCHIVE DESTINATION ${INSTALL_LIBDIR}
         COMPONENT SharedLibraries NAMELINK_SKIP)
-INSTALL(TARGETS libmariadb LIBRARY DESTINATION ${INSTALL_LIBDIR}
+INSTALL(TARGETS libmariadb
+        RUNTIME DESTINATION ${INSTALL_BINDIR}
+        LIBRARY DESTINATION ${INSTALL_LIBDIR}
+        ARCHIVE DESTINATION ${INSTALL_LIBDIR}
         COMPONENT Development NAMELINK_ONLY)
 ENDIF()
 
diff --git a/libmariadb/secure/schannel_certs.c b/libmariadb/secure/schannel_certs.c
index 1bcfb85..953c743 100644
--- a/libmariadb/secure/schannel_certs.c
+++ b/libmariadb/secure/schannel_certs.c
@@ -66,7 +66,7 @@ static DWORD get_last_error()
 #define FAIL(...) \
    do{\
      status = get_last_error();\
-     ma_format_win32_error(errmsg, errmsg_len, status, __VA_ARGS__);\
+     ma_format_win32_error(errmsg, errmsg_len, status, ##__VA_ARGS__);\
      goto cleanup;\
   } while (0)
 
